src/actions/submission.js

```diff
@@ -0,0 +1,42 @@
+import { log, logError } from '../logger';
+
+import { GET_SUBMISSION_SUCCESS } from '../constants';
+
+export function getSubmissionSuccess(submission) {
+  return {
+    type: GET_SUBMISSION_SUCCESS,
+    payload: submission,
+  };
+}
+
+export const fetchSubmission = id => async (dispatch, getStore, { graphqlRequest }) => {
+  const query = `{
+      submission(id: "${id}") {
+        id
+        name
+        email
+        phone
+        date
+        summary
+        description
+        askAmount
+        totalCost
+      }
+    }`;
+  log('query is', query);
+
+  try {
+    const { data, errors } = await graphqlRequest(query);
+    log('got back', data, errors);
+
+    if (errors) {
+      return null;
```

Comment:
We might want to elaborate on this and show a error page rather than just redirecting to "/" (happens in the route).